### PR TITLE
add: rancher docs

### PR DESCRIPTION
**Pull request by ericpaulsen**
adds documentation for bringing up a Rancher cluster, with Coder-specific considerations.

feel free to suggest any necessary additions.

### CODE REVIEW

File: setup/kubernetes/rke.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+```console
+rke up --config cluster.yml
+```
+
+1. After the cluster is brought up, create your `kubeconfig` file and copy over the
+Rancher-generated configuration:
+
+```console
+mkdir -p $HOME/.kube/ && cp kube_config_cluster.yml $HOME/.test/config
```
Comment (johnstcn) [inline setup/kubernetes/rke.md:68]:
```suggestion
mkdir -p $HOME/.kube/ && cp kube_config_cluster.yml $HOME/.kube/config
```

Comment (ericpaulsen) [reply setup/kubernetes/rke.md:68]: ah yes

File: setup/kubernetes/rke.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+install a [Rancher-supported storage provisioner](https://rancher.com/docs/rancher/v2.5/en/cluster-admin/volumes-and-storage/provisioning-new-storage/#prerequisites).
+We recommend using [Longhorn](https://longhorn.io/), since it is tightly
+integrated with Rancher.
+
+### Networking
+
+Since the Coder default service is of `type: LoadBalancer`, we recommend
+[installing MetalLB into your Rancher cluster](https://metallb.universe.tf/installation/).
+You must install MetalLB prior to installing Coder for MetalLB to identify the
+Coder service and [expose it externally](https://metallb.universe.tf/usage/).
```
Comment (johnstcn) [inline setup/kubernetes/rke.md:33]: I recommend linking to Longhorn's recommended CNI providers: 

https://rancher.com/docs/rancher/v2.6/en/faq/networking/cni-providers/

File: setup/kubernetes/rke.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+- [Docker](https://docs.docker.com/engine/install/)
+- [Helm](https://helm.sh/docs/intro/install/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+- [rke](https://rancher.com/docs/rke/latest/en/installation/)
+
+### Storage
+
+Since Coder requires dynamic storage provisioning, you'll need to
+install a [Rancher-supported storage provisioner](https://rancher.com/docs/rancher/v2.5/en/cluster-admin/volumes-and-storage/provisioning-new-storage/#prerequisites).
```
Comment (johnstcn) [inline setup/kubernetes/rke.md:24]: Latest version is 2.6

```suggestion
install a [Rancher-supported storage provisioner](https://rancher.com/docs/rancher/v2.6/en/cluster-admin/volumes-and-storage/provisioning-new-storage/#prerequisites).
```

File: setup/kubernetes/rke.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+description: Learn how to setup a Rancher cluster for your Coder deployment
+---
+
+This deployment guide shows you how to set up a bare-metal Rancher Kubernetes
+Engine (RKE) cluster on which Coder can deploy.
+
+## Prerequisites
+
+You must have at least one Linux host (node) with the following utilities
+installed:
```
Comment (johnstcn) [inline setup/kubernetes/rke.md:12]: We should link the Rancher docs on requirements:

https://rancher.com/docs/rke/latest/en/os/

**Review by johnstcn — APPROVED** [review]
👍 🤠 🐄

**Pull request merged**